sdk/java-sdk/src/main/scala/com/akkaserverless/javasdk/impl/effect/SecondaryEffectImpl.scala

```diff
@@ -43,21 +40,20 @@ sealed trait SecondaryEffectImpl {
           ClientAction(
             ClientAction.Action
               .Failure(com.akkaserverless.protocol.component.Failure(commandId, failure.description))))
-      case _: effect.NoReply[_] @unchecked | effect.NoSecondaryEffectImpl =>
-        if (allowNoReply) {
-          None
-        } else {
-          throw new RuntimeException("No reply or forward returned by command handler!")
-        }
+      case NoSecondaryEffectImpl =>
+        throw new RuntimeException("No reply or forward returned by command handler!")
     }
   }
 }
 
 case object NoSecondaryEffectImpl extends SecondaryEffectImpl {
   override def sideEffects: Vector[SideEffect] = Vector.empty
 
-  override def addSideEffects(sideEffects: Iterable[SideEffect]): SecondaryEffectImpl =
-    NoReply(sideEffects.toVector)
+  override def addSideEffects(newSideEffects: Iterable[SideEffect]): SecondaryEffectImpl =
```

Comment:
I'm not sure what to do with NoSecondaryEffectImpl itself. Is the default value for `_secondaryEffect` in `EventSourceEntityEffectImpl` for example

` private var _secondaryEffect: SecondaryEffectImpl = NoSecondaryEffectImpl`

Reply:
I think it'd make more sense to make that an option/optional in the internal effect builder impls now that it is not possible to attach side effects to a no-reply, but maybe I'm missing something?